lib/pbench/server/database/models/datasets.py

```diff
@@ -980,6 +986,19 @@ def create(**kwargs) -> "Metadata":
         else:
             return meta
 
+    @staticmethod
+    def get_native_key(key: str) -> str:
+        """
+        Extract the root key name
+
+        Args:
+            key:    Key path (e.g., "user.tag")
+
+        Returns:
+            native SQL key name ("user")
+        """
+        return key.lower().split(".").pop(0)
```

Comment:
Why `pop()`?  I think the result of `split()` is a temporary list...there's no need to modify it, is there?  I.e., why not just:
```
        return key.lower().split(".")[0]
```

Reply:
I moved the code; in the original context there was a reason, but the historical artifact has just been dragged along...